### DISCUSSION TITLE
a Quick Start example for MDI

### DISCUSSION THREAD

**Discussion by EdoFro** [EdoFro/Freeplane_Map-Drive-Inator#39] [General]
Hi,
I want to improve the documentation for this add-on, but I never find time to do it (I really just don't like writing manuals and documentation, so I procastinate it)

I will try to write a short Quickstarting guide here. Hope this helps.

<details><summary><h4>1. create a new mind map</h4></summary>

File->New map (Ctrl+N)

</details>

<details><summary><h4>2. save the map (this is important if you want to use relative links for your file structure)</h4></summary>

File->Save map as… (Ctrl+Shift+S)

</details>

<details><summary><h4>3. copy/paste (or drag and drop) a folder link into your mind map</h4></summary>

example drag&droping the folder from Windows explorer into Freeplane:

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/fca5e17e-bb33-4da2-b71c-d1cb6f3c502e)

</details>

<details><summary><h4>4. Import MDI styles into your map</h4></summary>

Edo Tools->MapDriveInator->Import MDI styles

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/69773d5f-2155-43ab-9be6-60dcac393c3a)

</details>

<details><summary><h4>5. if the folder structure you want to import is big and you want to make MDI to ignore some subfolders then continue with the following steps.</h4></summary>

If not, you can go directly to step 13.

</details>

<details><summary><h4>6. Select the folder node and click on 'create/update base folder node'. </h4></summary>

Edo Tools->MapDriveInator->Create/update base folder node

This command creates the base folder node but doesn't import (or update) its files and subfolders.

</details>

<details><summary><h4>7. When creating the baseFolderNode, define its parameters as you like or use its standard values.</h4></summary>

Dialogs will show up to help you defining its values

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/ac81856f-60b4-4e95-90d9-dd3f48c25a66)

</details>

<details><summary><h4>8. if the folder structure to import is big, then you can change the maxDepth value so it imports only a few levels.</h4></summary>

In this example I set it to 0 to get just the first level of folders

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/d1fcee6a-6cf7-4e1e-a5c7-218b3e45bd9d)

</details>

<details><summary><h4>9. use the 'import just folders' command</h4></summary>

Edo Tools->MapDriveInator->Import just folders

in this example it imports the first level of folders

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/3bd67e5d-06cd-4621-a2dd-3b042f3bd708)

</details>

<details><summary><h4>10. assign the 'locked' style to the branches you want MDI to skip (ignore)</h4></summary>

create a node, assign it the 'locked' style and move the folders you want to ignore into it.

Or assign that style directly to the folders you want to skip.

In the example I used both strategies

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/fd8c4dee-17ea-40da-9e45-d27477dea13d)

</details>

<details><summary><h4>11. change the maxDepth to import more folder levels and repeat steps 9 and 10</h4></summary>

In the example I set the MaxDepth to -1 (no maxDepth restriction) and imported just folders again

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/10c2ec18-01ed-43c9-963f-341d7c127372)

</details>

<details><summary><h4>12. once you are ready defining the branches to ignore continue with the next step</h4></summary>

I set new subfolders to be ignored:

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/fea7aaec-0c87-4df3-99a4-90fd2f7312e6)

</details>

<details><summary><h4>13. use command "Map drive inator" to import all the files</h4></summary>

Edo Tools->MapDriveInator->Map drive inator

Now it imports all the files into the map.

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/7f668d7f-dcca-4f55-b53b-3f23f62fefa6)

the new files appear in the "new imported files" node. this way you can see in one place all the new files every time you use "Map drive inator" command.

This is useful when working with big maps.

you can add comments in their notes or details, or add childnodes with additional information before distributing them to their corresponding map positions.

</details>

<details><summary><h4>14. use command "Distribute new nodes" to place the new imported files in their corresponding Mindmap's branches</h4></summary>

Edo Tools->MapDriveInator->Distribute new nodes

the new file nodes are moved to their corresponding map positions and "marked" as new, this way you can easily spot them in the map and move them into their exact position as you like.

![image](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/35700575/754d901e-1c5b-4983-bca1-bfe843334928)

</details>

Use the commands in the "Clear formats" menu to erase the "freshNew" format from the nodes in each selected branch.

That's it!
hope it helps!

BR
edo

**Comment by EdoFro** [comment]
Hi all,
If you have any comment or question about this quick guide, please don't hesitate to write it here.

I want to include this in the documentation of the add-on, so if you:
 - discover any error in the steps
 - have an idea how to improve the writing
 - think that an additional image is needed
 - think that some images may be improved
 - etc
, please let me know.

Thanks in advance,
edo

**Comment by ChuanShuoJiPinQiong** [comment]
I double clicked "MapDriveInator-v0.0.13.addon.mm"  and installed the addon.
I've tried v1.11.13、1.11.14、1.10.6 Freeplane, and all of them got the same error when execute "import MDI styles" on the node with the folder icon (the node with folder icon seems to be generated after drag and drop the folder on its parent node).
The Message shows:
![error when “import MDI styles”](https://github.com/EdoFro/Freeplane_Map-Drive-Inator/assets/63838616/0abbb005-0339-405c-bea3-183b2c6b3c6d)

Thanks for the great work.

**Reply by ChuanShuoJiPinQiong** [reply]
I installed the pseudoFreeplaneAPI addon and no error any more. Thanks for the great work :) .

**Comment by Pierre-de-la-Verre** [comment]
Hi,
what's the current state of this project? Is it working? Working fine? Is the documentation correct? What's about the "pseudoFreeplaneAPI addon"?

Thanks for reply!

**Reply by EdoFro** [reply]
Hi,
I use it daily and it works fine for me. But I think I've made some modifications that aren't here on GitHub.
I'll have to check and comment later.

**Reply by Pierre-de-la-Verre** [reply]
Would be fine - thanks!

**Comment by acebone** [comment]
Hi EdoFro

I would really like to check out this script, but when I try to import MDI styles I get this error

<img width="739" height="290" alt="image" src="https://github.com/user-attachments/assets/505ef626-67cb-465c-b691-b42f12f253b9" />
